src/app/modules/auth/auth.utils.ts

```diff
@@ -63,3 +74,60 @@ export const getUserIdFromSession = (
 ): string | undefined => {
   return session?.user?._id as string | undefined
 }
+
+/**
+ * Maps errors from creating a redirect URL to status codes and error messages to return to frontend.
+ * @param error
+ */
+export const mapRedirectUrlError: MapRouteError = (
```

Comment:
Same here, this whole function should be in public form utils; and should we also rename it to `mapFormAuthRedirectRouteError`.

Actually, is there a reason why we do not use the usual `mapRouteError`?

Reply:
i think the main motivating factor for not using the `mapRouteError` was the fear that there might be identical errors that are handled differently. i think this is premature  on my end and i'll combine it with the appropriate `mapRouteError` in public-forms

Reply:
I separated the two mapRouteError functions in the same module because it was easier to check that the correct errors were being covered for each one. not sure if there's any benefit to smooshing all the errors together into a gigantic switch statement even if they're for different endpoints.

Reply:
i do think that if we want `mapRouteError` to just be a black box and let consumers call and get the appropriate error code/message, it should just be 1 huge switch case HAHA

after checking, i don't think there are any errors that we treat differently for redirect, so i combined it into public-form's `mapRouteError`

Reply:
>if we want mapRouteError to just be a black box and let consumers call and get the appropriate error code/message

I don't understand your point. we do want this - that's what a helper function is for - but why combine the switch cases for unrelated endpoints?

Reply:
wouldn't the value of `mapRouteError` be for the module where it's rooted in - i.e., that for all routes rooted at `/forms`, we treat all errors similarly and defer the appropriate response to `mapRouteError`?

Reply:
it's a minor thing so I'm not going to belabour this discussion anymore, but I don't think it's the right approach to anyhow throw unrelated errors together in the same switch case. it makes it so much harder to check that the errors are being covered exhaustively.

Reply:
@seaerchin i defer to anti, you decide if you want to extract out the mapXError function to a specific form auth one. Fwiw I also agree with him

Reply:
ok sure! will do